far/vmenu.cpp

```diff
@@ -1957,32 +2054,34 @@ void VMenu::DisplayObject()
 
 	HideCursor();
 
+	const auto BoxType{ CalculateBoxType() };
+
 	if (!CheckFlags(VMENU_LISTBOX) && !SaveScr)
 	{
-		if (!CheckFlags(VMENU_DISABLEDRAWBACKGROUND) && !(m_BoxType==SHORT_DOUBLE_BOX || m_BoxType==SHORT_SINGLE_BOX))
+		if (!CheckFlags(VMENU_DISABLEDRAWBACKGROUND) && !(BoxType==SHORT_DOUBLE_BOX || BoxType==SHORT_SINGLE_BOX))
 			SaveScr = std::make_unique<SaveScreen>(rectangle{ m_Where.left - 2, m_Where.top - 1, m_Where.right + 4, m_Where.bottom + 2 });
 		else
 			SaveScr = std::make_unique<SaveScreen>(rectangle{ m_Where.left, m_Where.top, m_Where.right + 2, m_Where.bottom + 1 });
 	}
 
 	if (!CheckFlags(VMENU_DISABLEDRAWBACKGROUND) && !CheckFlags(VMENU_LISTBOX))
 	{
-		// BUGBUG, dead code
+		// BUGBUG, dead code -- 2023-07-08 MZK: Is it though? I've got here when pressed hotkey of "Select search area" combobox on "find file" dialog.
```

Comment:
TBH I don't remember the reason of the comment.

Maybe it applies only to the `!CheckFlags(VMENU_DISABLEDRAWBACKGROUND)` part of the condition, which is always true, since nothing sets the flag anywhere (perhaps we should remove it).
Or maybe I didn't notice the `!` and thought that the whole branch is unreachable.

Reply:
Interesting. I adjusted the comment.

If this this flag is indeed effectively unused, can we repurpose it? I need a new flag for enabling vertical alignment of annotations feature. It should be enabled only for specific lists, not for regular vertical menus, for example. Unfortunately, `VMENU_FLAGS` utilize all 32 bits, so I was not sure how to fit another one.

Reply:
Can't see why not (with a new name obviously).

Reply:
Thanks. I thought about plugin interface which might be affected. I am always uncertain about plugin API.

Leaving it for now, and will get back to it when implementing vertical alignment of annotations.